### DISCUSSION TITLE
Color formatting for source (RPGLE)

### DISCUSSION THREAD

**Discussion by NextPage66** [codefori/vscode-ibmi#2178] [General]
I am using the most current version of Code for IBM i.
We are still learning how to use this editor but all of our code was created and maintained in SEU.   
When we make changes in our source we will mark our initials in columns 1-5.
We are starting to maintain source in Code for IBM i.
I have noticed the color of the line typically changes to light grey when it is commented out and nothing is in columns 1-5.
If there is anything in columns 1-5 and we comment the line out, it will turn to black(and possibly other colors for op-codes).
Why doesn't it change to light grey like the others? 
Is there something special about columns 1-5 in the editor?

Sometimes there are large blocks of code that are light grey and they are not commented out.
Looks like the standard color for source is green. But sometimes commented lines will also stay green.
I am confused on this coloring and it is hard to read.
CL source seems to be ok.
DSPF source seem to be ok.
But SQLRPGLE has colors i do not understand what is going on.

I have tried different themes.  Currently using Quiet Light Theme.
Is there a menu option that describes formatting rules?
Is there a special rule for columns 1-5?

Thanks for any help.

**Comment by chrjorgensen** [comment]
@NextPage66 You need to provide some screenshots and code samples, so we can see and reproduce the errors...

**Reply by NextPage66** [reply]
![colors-5](https://github.com/user-attachments/assets/c10531d2-560d-4b51-a132-55c15ea88267)
![colors-4](https://github.com/user-attachments/assets/29a11a0b-2626-4a80-bf1b-2954d8155aef)
![colors-3](https://github.com/user-attachments/assets/803a657d-82e2-4775-8431-402a30073d9b)
![colors-2](https://github.com/user-attachments/assets/21fe88d2-7b6f-49c3-8326-d91b3c5fee01)
![colors-1](https://github.com/user-attachments/assets/a89c25bc-2591-46fa-b102-98563b820f18)

**Reply by worksofliam** [reply]
@NextPage66 Can you share a list of extensions you have installed? Thanks!

**Reply by NextPage66** [reply]
Is there a quick way to get the list rather than just print screens.  I have 36 extensions.

**Reply by sebjulliand** [reply]
Click on `Report an issue` here
![image](https://github.com/user-attachments/assets/9d29306d-b627-40bf-87b3-cb42e5d9d675)

It will open an issue template on GitHub pre-filled with various information.
Don't submit it. Just copy and paste what's under this section:
![image](https://github.com/user-attachments/assets/7345d280-de59-4ca8-92f7-136dba02e7b2)

**Reply by NextPage66** [reply]
thanks.

**Reply by NextPage66** [reply]
Extension|Author (truncated)|Version
---|---|---
ibmi-languages|bar|0.6.17
cobol|bit|10.7.9
vscode-rpgfree|Bri|0.0.32
vscode-firefox-debug|fir|2.9.10
code-for-ibmi|hal|2.12.0
ibm-i-development-pack|hal|0.1.1
vscode-db2i|hal|1.0.0
vscode-displayfile|hal|0.1.2
vscode-ibmi-walkthroughs|hal|0.5.0
vscode-rpgle|hal|0.26.8
vscode-test-explorer|hbe|2.21.1
db2connect|IBM|2.3.1
ibmidebug|IBM|2.0.1
vscode-clle|IBM|1.1.7
vscode-python-test-adapter|lit|0.8.2
data-workspace-vscode|ms-|0.5.0
mssql|ms-|1.22.1
sql-bindings-vscode|ms-|0.4.0
sql-database-projects-vscode|ms-|1.4.1
debugpy|ms-|2024.8.0
python|ms-|2024.10.0
vscode-pylance|ms-|2024.7.1
remote-ssh|ms-|0.112.0
remote-ssh-edit|ms-|0.86.0
notepadplusplus-keybindings|ms-|1.0.7
remote-explorer|ms-|0.4.3
test-adapter-converter|ms-|0.1.9
sqltools|mtx|0.28.3
rpg|Nie|0.2.1
vscode-print|pdc|0.12.16
geo-data-viewer|Ran|2.6.0
cllepkg|Spi|1.0.0
errorlens|use|3.20.0
vscode-todo-highlight|way|1.0.5

**Reply by chrjorgensen** [reply]
@NextPage66 The syntax colour coding for IBM i source types like RPGLE, CLLE etc. is provided by the [vscode-ibmi-languages](https://github.com/barrettotte/vscode-ibmi-languages) extension - not by Code for IBM i.

Please raise an issue over there and we will handle it in the right place. Also please provide your code samples as text - this will make it much easier to test your issues and any fixes developed. Thanks! :+1:

**Comment by NextPage66** [comment]
i did thanks
